flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SchedulerUtils.java

```diff
@@ -64,34 +61,6 @@ public static CompletedCheckpointStore createCompletedCheckpointStoreIfCheckpoin
         }
     }
 
-    @VisibleForTesting
-    static CompletedCheckpointStore createCompletedCheckpointStore(
```

Comment:
What is the intuition behind moving this into CheckpointRecoveryFactory?

Reply:
The intuition was that I wanted to reuse the `CheckpointsCleaner` instance in the cleanup of the `CompletedCheckpointsStore`. But that's not possible as is because the `CheckpointsCleaner` is closed on the `JobMaster`. Instead, we should use a `CheckpointsCleanerFactory`, instead, that is passed into the different components that need a `CheckpointsCleaner` instance.